packages/flutter_tools/lib/src/android/gradle_errors.dart

```diff
@@ -503,20 +503,30 @@ final GradleHandledError minCompileSdkVersionHandler = GradleHandledError(
   eventLabel: 'min-compile-sdk-version',
 );
 
+final RegExp _agpJavaError = RegExp(r'Android Gradle plugin requires Java\s+\d+\s+to run');
+
 @visibleForTesting
-final GradleHandledError jvm11RequiredHandler = GradleHandledError(
+final GradleHandledError incompatibleJavaAndAgpVersionsHandler= GradleHandledError(
   test: (String line) {
-    return line.contains('Android Gradle plugin requires Java 11 to run');
+    return _agpJavaError.hasMatch(line);
   },
   handler: ({
     required String line,
     required FlutterProject project,
     required bool usesAndroidX,
   }) async {
+    final File settingsFile = project.directory
+      .childDirectory('android')
+      .childFile('settings.gradle');
     globals.printBox(
-      '${globals.logger.terminal.warningMark} You need Java 11 or higher to build your app with this version of Gradle.\n\n'
-      'To get Java 11, update to the latest version of Android Studio on https://developer.android.com/studio/install.\n\n'
-      'To check the Java version used by Flutter, run `flutter doctor -v`.',
+      '${globals.logger.terminal.warningMark} Your version of Java is incompatible with your project\’s Android Gradle Plugin version.\n\n'
```

Comment:
Let's  rearrange the order here a bit to be the most helpful to the user. 

Good start, tells the user what is wrong immediately. It could be made better by saying what version was expected vs what they are using. 
`Your version of Java is incompatible with your project\’s Android Gradle Plugin version.\n\n'` 

`To fix ...` this is the next step the user will take. Likely without reading the rest of the error. 

`If that does not work ...` I dont think they can set the correct jdk until they know what version they need to have. 

Recommendation: 
Give the error including the versions. Which could be printing the line that was evaluated since android actually has a good error here. 
Then tell user how to check the version of java flutter is using. 
Then say how to check the version they need. 
Then say something like "try updating to the latest android sdk and android studio" even though android studio is not a requirement. 
Then say how to set the version of java flutter uses. 

I think the recommended order is mostly the same information but makes it less likely someone makes their problem worse by setting bad version of java.

Reply:
Discussed with @gmackall and decided to include the gradle thrown error line (which includes the versions) and remove the directions to check/find the required version of Java. This is because the gradle thrown error already indicates the required version of Java.

Reply:
Consider including where the find the compatible versions in a comment then. Even if it only helps ourselves.

Reply:
done